Misc/NEWS.d/next/Core_and_Builtins/2025-07-28-17-01-05.gh-issue-88886.g4XFPb.rst

```diff
@@ -0,0 +1,3 @@
+The codecs lookup function now performs only minimal normalization of the
```

Comment:
I'd write "The codecs lookup function now again performs only minimal normalization of the encoding name before passing it to the search functions: all ASCII letters are converted to lower case, spaces are replaced with hyphens. This restores the pre-Python 3.9 behavior.".

Reply:
Yes, it restores the pre-Python 3.9 behavior. I wonder if we can go further -- do not replace spaces, or replace them with underscores instead of hyphens, or replace also hyphens or underscores. We can handle this in `normalize_encoding()` or just add aliases. `iconv` on FreeBSD do not convert any non-letter character. `iconv` on Linux preserves alphanumerics and "_-.,:/", and removes all others.

Should it still convert to lower case? Most standard encoding names in the IANA registry are in upper case. `iconv` encoding names are all in upper case (on Linux and FreeBSD). But this is potentially a breaking change without much benefit.

Reply:
No, let's not change the behavior again. It's been like this for ages before the Python 3.9 change and we really only want to roll back that change.

The lower casing was added to make sure that encoding names are handled case insensitive. It's rather common to default to lower casing rather then upper casing to do this.

Reply:
@serhiy-storchaka Could you fix the sentence to what I wrote above ? The PR should be ready for merging then.

Reply:
Hmm, Github now shows the change... weird interface.